pkg/telemetry/client_test.go

```diff
@@ -127,14 +129,21 @@ func TestNewClient_PacketSizeIsZero(t *testing.T) {
 }
 
 func TestRun_ReportNoCalled(t *testing.T) {
```

Comment:
I'm not so sure about this test, because I think the more important part is about context cancellation stopping the Running rather than not sending data when having an error.